### ISSUE TITLE
build/run time initialization config of software.amazon.awssdk.crt. classes breaks native build

### ISSUE THREAD

**Issue by thomascube** [quarkiverse/quarkus-amazon-services#1086]
When building the latest version 2.10.0 in a Quarkus native application, I get the following errors which break the build:

```
Error: Classes that should be initialized at run time got initialized during image building:
 software.amazon.awssdk.crt.CRT the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'CRT.class'). software.amazon.awssdk.crt.checksums.CRC32C caused initialization of this class with the following trace: 
        at software.amazon.awssdk.crt.CRT.<clinit>(CRT.java:38)
        at software.amazon.awssdk.crt.checksums.CRC32C.<clinit>(CRC32C.java:15)

software.amazon.awssdk.crt.Log the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'Log.class'). software.amazon.awssdk.crt.checksums.CRC32C caused initialization of this class with the following trace: 
        at software.amazon.awssdk.crt.Log.<clinit>(Log.java:16)
        at software.amazon.awssdk.crt.CRT.<clinit>(CRT.java:70)
        at software.amazon.awssdk.crt.checksums.CRC32C.<clinit>(CRC32C.java:15)

com.oracle.svm.core.util.UserError$UserException: Classes that should be initialized at run time got initialized during image building:
 software.amazon.awssdk.crt.CRT the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'CRT.class'). software.amazon.awssdk.crt.checksums.CRC32C caused initialization of this class with the following trace: 
        at software.amazon.awssdk.crt.CRT.<clinit>(CRT.java:38)
        at software.amazon.awssdk.crt.checksums.CRC32C.<clinit>(CRC32C.java:15)

software.amazon.awssdk.crt.Log the class was requested to be initialized at run time (from feature io.quarkus.runner.Feature.beforeAnalysis with 'Log.class'). software.amazon.awssdk.crt.checksums.CRC32C caused initialization of this class with the following trace: 
        at software.amazon.awssdk.crt.Log.<clinit>(Log.java:16)
        at software.amazon.awssdk.crt.CRT.<clinit>(CRT.java:70)
        at software.amazon.awssdk.crt.checksums.CRC32C.<clinit>(CRC32C.java:15)
```

This output is shown when adding the `--trace-class-initialization=software.amazon.awssdk.crt.Log,software.amazon.awssdk.crt.CRT` option. However, setting `--initialize-at-run-time=software.amazon.awssdk.crt.checksums.CRC32C` doesn't solve the issue. The build still breaks but even with `--trace-class-initialization=software.amazon.awssdk.crt.checksums.CRC32C` I see no further information that would help me fixing the build.

Please advise.

Quarkus version: 3.6.5
Java compiler version: 17
Native build done with docker support (`-Pnative -Dquarkus.native.container-build=true`)

```
[INFO] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Running Quarkus native-image plugin on native-image 17.0.9 2023-10-17
OpenJDK Runtime Environment Mandrel-23.0.2.1-Final (build 17.0.9+9)
OpenJDK 64-Bit Server VM Mandrel-23.0.2.1-Final (build 17.0.9+9, mixed mode)
```

**Comment by scrocquesel** [comment]
Thanks for reporting this. As a quick workaround, you can stay on 2.9.1.

Meanwhile, could you provide the content of your pom.xml or a small reproducer ?

**Comment by scrocquesel** [comment]
Would you mind testing with the version from the linked PR?

I widened types that should be runtime-initialized, but not knowing why CRC32 is initialized in your case and not in our integration tests, I'm not confident with the fix.

**Comment by thomascube** [comment]
@scrocquesel Here's the pom.xml from your project:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.my.group</groupId>
  <artifactId>just-another-kubernetes-operator</artifactId>
  <name>just-another-kubernetes-operator</name>
  <version>0.0.6-SNAPSHOT</version>
  <packaging>jar</packaging>
  <properties>
    <maven.compiler.parameters>true</maven.compiler.parameters>
    <maven.compiler.source>17</maven.compiler.source>
    <maven.compiler.target>17</maven.compiler.target>
    <compiler-plugin.version>3.12.1</compiler-plugin.version>
    <quarkus.version>3.6.6</quarkus.version>
    <quarkus-sdk.version>6.5.1</quarkus-sdk.version>
    <quarkus-amazon.version>2.10.0</quarkus-amazon.version>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    <commons-lang3.version>3.14.0</commons-lang3.version>
    <surefire-plugin.version>3.2.5</surefire-plugin.version>
  </properties>

  <dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.quarkiverse.operatorsdk</groupId>
        <artifactId>quarkus-operator-sdk-bom</artifactId>
        <version>${quarkus-sdk.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
      <dependency>
        <groupId>io.quarkiverse.amazonservices</groupId>
        <artifactId>quarkus-amazon-services-bom</artifactId>
        <version>${quarkus-amazon.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>io.quarkiverse.operatorsdk</groupId>
      <artifactId>quarkus-operator-sdk</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkiverse.operatorsdk</groupId>
      <artifactId>quarkus-operator-sdk-bundle-generator</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-micrometer-registry-prometheus</artifactId>
      <version>${quarkus.version}</version>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-config-yaml</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-logging-json</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkiverse.amazonservices</groupId>
      <artifactId>quarkus-amazon-iam</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkiverse.amazonservices</groupId>
      <artifactId>quarkus-amazon-s3</artifactId>
    </dependency>
    <dependency>
      <groupId>software.amazon.awssdk</groupId>
      <artifactId>url-connection-client</artifactId>
    </dependency>
    <dependency>
      <groupId>software.amazon.awssdk</groupId>
      <artifactId>netty-nio-client</artifactId>
    </dependency>
    <dependency>
      <groupId>software.amazon.awssdk</groupId>
      <artifactId>aws-crt-client</artifactId>
    </dependency>
    <dependency>
      <groupId>software.amazon.awssdk</groupId>
      <artifactId>iam-policy-builder</artifactId>
      <version>2.23.4</version>
    </dependency>
    <dependency>
      <groupId>org.projectlombok</groupId>
      <artifactId>lombok</artifactId>
      <version>RELEASE</version>
      <scope>compile</scope>
    </dependency>
    <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-lang3</artifactId>
      <version>${commons-lang3.version}</version>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-test-kubernetes-client</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-junit5</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-junit5-mockito</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>org.awaitility</groupId>
      <artifactId>awaitility</artifactId>
      <version>4.2.0</version>
      <scope>test</scope>
    </dependency>
  </dependencies>

  <build>
    <plugins>
      <plugin>
        <groupId>io.quarkus</groupId>
        <artifactId>quarkus-maven-plugin</artifactId>
        <version>${quarkus.version}</version>
        <executions>
          <execution>
            <goals>
              <goal>build</goal>
            </goals>
          </execution>
        </executions>
      </plugin>
      <plugin>
        <artifactId>maven-compiler-plugin</artifactId>
        <version>${compiler-plugin.version}</version>
        <configuration>
          <compilerArgs>
            <arg>-parameters</arg>
          </compilerArgs>
        </configuration>
      </plugin>
      <plugin>
        <artifactId>maven-surefire-plugin</artifactId>
        <version>${surefire-plugin.version}</version>
        <configuration>
          <systemPropertyVariables>
            <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
          </systemPropertyVariables>
        </configuration>
      </plugin>
    </plugins>
  </build>

  <profiles>
    <profile>
      <id>native</id>
      <properties>
        <quarkus.package.type>native</quarkus.package.type>
      </properties>
    </profile>
  </profiles>

</project>
```

**Comment by thomascube** [comment]
> why CRC32 is initialized in your case

I can't tell why. The option `--trace-class-initialization=software.amazon.awssdk.crt.checksums.CRC32C` doesn't yield any output that would help. In our code we use `IamClient`and `S3Client` via constructor dependency injection.

**Comment by thomascube** [comment]
@scrocquesel I now tested with the linked PR version and the native build of our project succeeds.

**Comment by scrocquesel** [comment]
@thomascube shipped in 2.10.1
